admin/src/react-components/import-content.js

```diff
@@ -73,129 +80,353 @@ class ImportContentComponent extends Component {
     }
   }
 
-  async onSubmit(e) {
+  addImport(url, importUrl, type, asset, isDefault, isBase, isFeatured) {
+    const { imports } = this.state;
+    imports.push({
+      url,
+      importUrl,
+      type,
+      asset,
+      result: RESULTS.pending,
+      isDefault,
+      isBase,
+      isFeatured,
+      isImported: false,
+      isEnabled: true
+    });
+    this.setState({ imports });
+  }
+
+  setImportResult(url, result) {
+    this.setImportFields(url, i => (i.result = result));
+
+    if (result === RESULTS.new_listing || result === RESULTS.existing_listing || result === RESULTS.failed) {
+      this.setImportFields(url, i => (i.isImported = true));
```

Comment:
Yeah, the `isImported` field is basically a way to mark the row as no longer selectable. It ended up not being necessary, but my original idea was to make it so if you imported two URLs in the same page load it would append to the table.

Reply:
(whereas the final version here ends up just clearing the table each time)

Reply:
Oh I should correct that, this flag is also used as a way to determine if there are any non-completed rows as well.